### DISCUSSION TITLE
File-based Metadata API should support `favicon.svg`

### DISCUSSION THREAD

**Discussion by Svish** [vercel/next.js#51197] [Ideas]
### Goals

1. Use an SVG as the main `favicon` of my site.

### Background

Currently the file-based Metadata API only supports `.ico` for `favicon`. Browsers support SVG-icons, and the awesome thing about them is they can be responsive to preferences for `light` and `dark` themes.

### Proposal

![image](https://github.com/vercel/next.js/assets/142162/81dce93e-32c3-4bc1-ae7c-9be7b1c671e1)

**Comment by hesalx** [comment]
Browsers support SVG icons but browser do not download `favicon.svg` automatically. That is, you have to define `rel=icon` explicitly. But in that case the name does not matter. Actually, it can be any URL. Check this page's source, the GitHub's icon is SVG but isn't even from the same domain.

On the very screenshot you provided it states that SVG is supported as `icon.svg`. It results in 
```html
<link rel="icon" href="/icon.svg" type="image/svg+xml">
```
which is exactly what "Browsers support SVG-icons" means.

"favicon.ico" is just the name convention for implicitly loaded icons. The real deal is `rel=icon`.

**Reply by Svish** [reply]
Do you know if one _has_ to have that implicitly loaded icon on a website? What happens if it does not exist?

**Reply by hesalx** [reply]
The answer I can deduce is: No, one does not have to have it.

The static-path `/favicon.ico` is "Discouraged" for almost two decades now: https://www.w3.org/2005/10/howto-favicon

The global support of `png` icons is 97.17% which for all intents and purposes is 99.9% because the rest "Does not use favicons at all": https://caniuse.com/link-icon-png

The support of `png` icons includes support for `rel=icon` by definition, there is no `/favicon.png` fallback: https://html.spec.whatwg.org/multipage/links.html#rel-icon

Therefore, `rel=icon` without a static-path `/favicon.ico` is supported by 99.9% of internet users' browsers that support icons in the first place, according to [StatCounter GlobalStats](http://gs.statcounter.com/) (the source of https://caniuse.com/ stats).

The only edge case I can think of is if the user accesses the site for the first time (because icons are cached) and the first page they visit is a server error, or for other reason does not include `rel=icon`, the browser would still try to load `/favicon.ico`, but won't have `rel=icon` to rely on. Although in that case the missing icon is the least of the site owner's problems.

Some google services don't actually have `/favicon.ico`:
* https://search.google.com/favicon.ico – 404
* https://analytics.google.com/favicon.ico – 404

Even Can I Use doesn't have it:
* https://caniuse.com/favicon.ico - 404

**Reply by Svish** [reply]
Perfect, thank you for your answer. Think I'll just ignore the favicon then. 👍

**Reply by nickserv** [reply]
> Do you know if one _has_ to have that implicitly loaded icon on a website? What happens if it does not exist?

No, it's optional. If you don't have `<link rel="icon" href="...">`, the browser will request `/favicon.ico`, and ignore it if there's a 404 response. You can confirm this by using network devtools.

**Comment by DmitriyPast** [comment]
Same issue. Using .svg as the site icon simply doesn't work and using <link> returns 404
